Source/Libraries/Adapters/GrafanaAdapters/GrafanaDataSourceBase_AncillaryOperations.cs

```diff
@@ -231,6 +236,12 @@ private static Task ProcessSquareDistributionAsync<T>(List<DataSourceValueGroup<
         if (!double.TryParse(setting, out double zoom) || zoom <= 0.0D)
             throw new SyntaxErrorException("Square distribution \"zoom\" setting is negative, zero or not a valid number.");
 
+        if (!settings.TryGetValue("", out setting))
+            setting = "";
+
+        string groupBy = setting;
+
```

Comment:
This set of operations does nothing, group by is always an empty string. Presume you are looking to do something like this:
```c#
        if (!settings.TryGetValue("groupBy", out setting))
            setting = "";

        string groupBy = setting;
```